src/view/components/site-selector/dot-site-selector-component.ts

```diff
@@ -38,4 +42,44 @@ export class SiteSelectorComponent extends BaseComponent {
 
         });
     }
+
+    /**
+     * Filter the users displayed in the dropdown by comparing if
+     * the user name characters set on the drowpdown search box matches
+     * some on the user names set on the userlist variable loaded on the
+     * ngOnInit method
+     *
+     * @param event - The event with the query parameter to filter the users
+     */
+    filterSites(event): void {
+        this.filteredSitesResults = [];
+        for(let i = 0; i < this.sites.length; i++) {
+            let site = this.sites[i];
+            if(site.hostName.toLowerCase().indexOf(event.query.toLowerCase()) >= 0) {
+                this.filteredSitesResults.push({
+                    label: site.hostName,
+                    value: site.identifier,
+                });
+            }
+        }
+    }
+
+    /**
+     * Display all the existing login as users availables loaded on the
+     * userList variable initialized on the ngOnInit method
+     *
+     * @param event - The click event to display the dropdown options
+     */
+    handleSitesDropdownClick(event): void {
+        this.filteredSitesResults = [];
+        setTimeout(() => {
+            for(let i = 0; i < this.sites.length; i++) {
+                let site = this.sites[i];
+                this.filteredSitesResults.push({
+                    label: site.hostName,
+                    value: site.identifier,
+                });
+            }
+        }, 100);
+    }
```

Comment:
Yes, the timeout seems to to be required http://forum.primefaces.org/viewtopic.php?f=35&t=47931&p=148519&hilit=autocomplete#p148519